.github/workflows/infra-check.yml

```diff
@@ -1,7 +1,7 @@
 # Check if only infrastructure files are changed when infrastructure label is set.
 name: infrastructure-check
 on:
-  pull_request:
+  pull_request_target:
```

Comment:
Shouldn't we prefer `pull_request` for good reasons? I have a problem to find a good source that would back you up. Could you share some links with us, please?

Reply:
Here: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token

```
You can use the permissions key to add and remove read permissions for forked repositories, but typically you can't grant write access.
```

With combination of here: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflows-in-forked-repositories

And yes, it shouldn't be used because you can inject your code to the PR which then could be executed. However, you can't really change the workflows which are executed because these are executed from the target branch (not your PR).
That means that this check is fine. This workflow doesn't execute any code from the PR, it just invokes git commands and compare status of these. Another thing is that someone has to approve external contributor test run before the workflow is executed.
I think we should be safe here.

Reply:
Thank you!